### PR TITLE
store both webhook hash and clone hash per build

### PR DESCRIPTION
**Pull request by amirbey**
Each build in federalist should store the actual hash of the build when it was cloned.

### CODE REVIEW

File: api/services/GithubBuildStatusReporter.js [hunk 1]
```diff
@@ -77,10 +77,10 @@ const reportBuildStatus = (build) => {
   let options = {};
 
   return new Promise((resolve, reject) => {
-    if (!build || !build.commitSha) {
-      reject(new Error('Build or commit sha undefined. Unable to report build status'));
-    } else {
+    if (build && (build.webhookCommitSha || build.cloneCommitSha)) {
```
Comment (davemcorwin) [inline api/services/GithubBuildStatusReporter.js:80]: Is there a scenario when we would have `cloneCommitSha` without `webhookCommitSha` ?

File: api/services/GithubBuildStatusReporter.js [hunk 2]
```diff
@@ -98,7 +98,7 @@ const reportBuildStatus = (build) => {
       options = {
         owner: site.owner,
         repo: site.repository,
-        sha: build.commitSha,
+        sha: build.cloneCommitSha || build.webhookCommitSha,
```
Comment (davemcorwin) [inline api/services/GithubBuildStatusReporter.js:101]: Hmmm, how does this work right now? Does Github show the latest reported status OR the reported status for the latest commit included in the PR? (looking at the API docs, I wonder if they are using https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-the-combined-status-for-a-specific-reference which would seem to be just the latest) Before this change I could see this being a bit wonky in that we may report a status for a different (out of date) commit!

This seeeeeems fine?

Makes me think we *really* should build specific commits...

File: frontend/propTypes.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ export const BUILD = PropTypes.shape({
   state: PropTypes.string,
   error: PropTypes.string,
   branch: PropTypes.string,
-  commitSha: PropTypes.string,
+  webhookCommitSha: PropTypes.string,
```
Comment (davemcorwin) [inline frontend/propTypes.js:37]: you're sending BOTH values to the frontend, right?

**Review by davemcorwin — COMMENTED** [review]
Thanks @amirbey . I would have suggested NOT renaming the existing column just for the sake of churn/risk, but I also realize that even with the change, it's still not totally accurate since some of the builds are not triggered from webhooks. **IF** we are going to rename it, maybe something like `requestedCommitSha` would be more accurate, but I think it's safer to leave it as is and just create the new column.

File: api/services/GithubBuildStatusReporter.js (hunk 1 above)
Comment (amirbey) [reply api/services/GithubBuildStatusReporter.js:80]: point taken

File: test/api/requests/build.test.js [hunk 4]
```diff
@@ -179,54 +183,19 @@ describe('Build API', () => {
           })
           .then((response) => {
             validateAgainstJSONSchema('POST', '/build', 200, response.body);
-            return Build.findOne({
+            return Build.findAll({
```
Comment (davemcorwin) [inline test/api/requests/build.test.js:186]: hmmmm, I *think* `findOne` is the same things a `findAll` except it just returns only the first record...

**Review by davemcorwin — APPROVED** [review]
thanks @amirbey ! question about `findOne` and `findAll` below, but otherwise good!

File: test/api/requests/build.test.js (hunk 4 above)
Comment (amirbey) [reply test/api/requests/build.test.js]: changed

**Pull request merged**